modules/labs-react/search-form/lib/SearchForm.tsx

```diff
@@ -118,173 +117,318 @@ function getInputColors(theme: SearchThemeAttributes, isFocused?: boolean) {
   };
 }
 
-const formCollapsedBackground = colors.frenchVanilla100;
+const formCollapsedBackground = '#fff';
 
-const maxWidth = 480;
-const minWidth = 120;
-
-const StyledSearchForm = styled('form')<
-  Pick<SearchFormProps, 'isCollapsed' | 'rightAlign' | 'grow'> & Pick<SearchFormState, 'showForm'>
->(
-  {
+const searchFormStencil = createStencil({
+  vars: {
+    minWidth: px2rem(120),
+    maxWidth: px2rem(480),
+    height: system.space.x10,
+    searchInputBackground: '',
+    searchInputBackgroundFocus: '',
+    searchInputBackgroundHover: '',
+    searchInputColor: '',
+    searchInputColorFocus: '',
+    searchInputPlaceholderColor: '',
+    searchInputPlaceholderColorFocus: '',
+    searchInputBoxShadow: '',
+    searchInputBoxShadowFocus: '',
+  },
+  parts: {
+    searchContainer: 'search-form-container',
+    combobox: 'search-form-combobox',
+    closeButton: 'search-form-close-button',
+    searchField: 'search-form-field',
+    submitSearchIcon: 'search-form-submit-search-icon',
+    openSearchIcon: 'search-form-open-search-icon',
+    searchInput: 'search-form-input',
+  },
+  base: ({
+    minWidth,
+    maxWidth,
+    searchContainerPart,
+    height,
+    comboboxPart,
+    closeButtonPart,
+    searchFieldPart,
+    submitSearchIconPart,
+    openSearchIconPart,
+    searchInputPart,
+    searchInputBackground,
+    searchInputBackgroundFocus,
+    searchInputBackgroundHover,
+    searchInputColor,
+    searchInputColorFocus,
+    searchInputPlaceholderColor,
+    searchInputPlaceholderColorFocus,
+    searchInputBoxShadow,
+    searchInputBoxShadowFocus,
+  }) => ({
```

Comment:
lol. Maybe we should have a `parts` and `vars`. Large components feel much more boilerplate when `searchContainer` is repeated so many times.